cmd/operator-sdk/add/api.go

```diff
@@ -84,19 +142,144 @@ Example:
+	cfg := &input.Config{
+		AbsProjectPath: absProjectPath,
+	}
+	roleFiles := ansible.RolesFiles{Resource: *r}
+	roleTemplates := ansible.RolesTemplates{Resource: *r}
 
-	log.Infof("Generating api version %s for kind %s.", apiVersion, kind)
+	// update watch.yaml for the given resource r.
+	if err := ansible.UpdateAnsibleWatchForResource(r, absProjectPath); err != nil {
+		return fmt.Errorf("failed to update the Watch manifest for the resource (%v, %v): (%v)",
+			r.APIVersion, r.Kind, err)
+	}
+
+	s := &scaffold.Scaffold{}
+	err = s.Execute(cfg,
+		&scaffold.CR{Resource: r},
+		&ansible.RolesReadme{Resource: *r},
+		&ansible.RolesMetaMain{Resource: *r},
+		&roleFiles,
+		&roleTemplates,
+		&ansible.RolesVarsMain{Resource: *r},
+		&ansible.RolesDefaultsMain{Resource: *r},
+		&ansible.RolesTasksMain{Resource: *r},
+		&ansible.RolesHandlersMain{Resource: *r},
+	)
+	if err != nil {
+		return fmt.Errorf("API scaffold failed: %v", err)
+	}
+	if err = genutil.GenerateCRDNonGo("", *r); err != nil {
+		return err
+	}
+	// update deploy/role.yaml for the given resource r.
+	if err := scaffold.UpdateRoleForResource(r, absProjectPath); err != nil {
+		return fmt.Errorf("failed to update the RBAC manifest for the resource (%v, %v): (%v)",
+			r.APIVersion, r.Kind, err)
+	}
+	return nil
+
+}
+
+func doHelmAPIScaffold() error {
+	// Create and validate new resource.
+	r, err := scaffold.NewResource(apiVersion, kind)
+	if err != nil {
+		return err
+	}
+	absProjectPath := projutil.MustGetwd()
+	cfg := &input.Config{
+		AbsProjectPath: absProjectPath,
+	}
+	// update watch.yaml for the given resource r.
+	if err := helm.UpdateHelmWatchForResource(r, absProjectPath); err != nil {
+		return fmt.Errorf("failed to update the Watch manifest for the resource (%v, %v): (%v)",
+			r.APIVersion, r.Kind, err)
+	}
+
+	createOpts := helm.CreateChartOptions{
+		ResourceAPIVersion: apiVersion,
+		ResourceKind:       kind,
+		Chart:              helmChartRef,
+		Version:            helmChartVersion,
+		Repo:               helmChartRepo,
+	}
+
+	r, chart, err := helm.CreateChart(cfg.AbsProjectPath, createOpts)
+	if err != nil {
+		return fmt.Errorf("failed to create helm chart: %v", err)
+	}
+
+	valuesPath := filepath.Join("<project_dir>", helm.HelmChartsDir, chart.Name(), "values.yaml")
+
+	rawValues, err := yaml.Marshal(chart.Values)
+	if err != nil {
+		return fmt.Errorf("failed to get raw chart values: %v", err)
+	}
+	crSpec := fmt.Sprintf("# Default values copied from %s\n\n%s", valuesPath, rawValues)
+	s := &scaffold.Scaffold{}
+	err = s.Execute(cfg,
+		&scaffold.CR{
+			Resource: r,
+			Spec:     crSpec,
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("helm scaffold failed: %v", err)
+	}
+	if err = genutil.GenerateCRDNonGo("", *r); err != nil {
+		return err
+	}
+	if err := scaffold.UpdateRoleForResource(r, absProjectPath); err != nil {
+		return fmt.Errorf("failed to update the RBAC manifest for resource (%v, %v): %v",
+			r.APIVersion, r.Kind, err)
+	}
```

Comment:
For Helm, we [generate a role from the Helm chart during `new`](https://github.com/operator-framework/operator-sdk/blob/179d8d6c81f8a14ff386f7d71372859551f56162/cmd/operator-sdk/new/cmd.go#L385-L392).

We need to be able to do this while updating the role as well.